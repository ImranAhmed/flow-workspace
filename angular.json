{
  "version": 1,
  "projects": {
    "flow-web-app": {
      "projectType": "application",
      "schematics": {
        "@nrwl/angular:component": {
          "style": "scss"
        }
      },
      "root": "apps/flow-web-app",
      "sourceRoot": "apps/flow-web-app/src",
      "prefix": "flow-web-app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/apps/flow-web-app",
            "index": "apps/flow-web-app/src/index.html",
            "main": "apps/flow-web-app/src/main.ts",
            "polyfills": "apps/flow-web-app/src/polyfills.ts",
            "tsConfig": "apps/flow-web-app/tsconfig.app.json",
            "aot": true,
            "assets": [
              "apps/flow-web-app/src/favicon.ico",
              "apps/flow-web-app/src/assets",
              {
                "glob": "**/*",
                "input": "libs/assets/src/lib/images",
                "output": "./assets/images"
              }
            ],
            "stylePreprocessorOptions": {
              "includePaths": ["libs/assets/src/lib/styles/"]
            },
            "styles": [],
            "scripts": []
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "apps/flow-web-app/src/environments/environment.ts",
                  "with": "apps/flow-web-app/src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "6kb",
                  "maximumError": "10kb"
                }
              ]
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "flow-web-app:build"
          },
          "configurations": {
            "production": {
              "browserTarget": "flow-web-app:build:production"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "flow-web-app:build"
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["apps/flow-web-app/tsconfig.app.json", "apps/flow-web-app/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!apps/flow-web-app/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "apps/flow-web-app/jest.config.js",
            "tsConfig": "apps/flow-web-app/tsconfig.spec.json",
            "passWithNoTests": true,
            "setupFile": "apps/flow-web-app/src/test-setup.ts"
          }
        }
      }
    },
    "flow-web-app-e2e": {
      "root": "apps/flow-web-app-e2e",
      "sourceRoot": "apps/flow-web-app-e2e/src",
      "projectType": "application",
      "architect": {
        "e2e": {
          "builder": "@nrwl/cypress:cypress",
          "options": {
            "cypressConfig": "apps/flow-web-app-e2e/cypress.json",
            "tsConfig": "apps/flow-web-app-e2e/tsconfig.e2e.json",
            "devServerTarget": "flow-web-app:serve"
          },
          "configurations": {
            "production": {
              "devServerTarget": "flow-web-app:serve:production"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["apps/flow-web-app-e2e/tsconfig.e2e.json"],
            "exclude": ["**/node_modules/**", "!apps/flow-web-app-e2e/**"]
          }
        }
      }
    },
    "ui-kit": {
      "projectType": "library",
      "root": "libs/ui-kit",
      "sourceRoot": "libs/ui-kit/src",
      "prefix": "ui-kit",
      "architect": {
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/ui-kit/tsconfig.lib.json", "libs/ui-kit/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/ui-kit/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/ui-kit/jest.config.js",
            "tsConfig": "libs/ui-kit/tsconfig.spec.json",
            "passWithNoTests": true,
            "setupFile": "libs/ui-kit/src/test-setup.ts"
          }
        },
        "storybook": {
          "builder": "@nrwl/storybook:storybook",
          "options": {
            "uiFramework": "@storybook/angular",
            "port": 4400,
            "config": {
              "configFolder": "libs/ui-kit/.storybook"
            }
          },
          "configurations": {
            "ci": {
              "quiet": true
            }
          }
        },
        "build-storybook": {
          "builder": "@nrwl/storybook:build",
          "options": {
            "uiFramework": "@storybook/angular",
            "outputPath": "dist/storybook/ui-kit",
            "config": {
              "configFolder": "libs/ui-kit/.storybook"
            }
          },
          "configurations": {
            "ci": {
              "quiet": true
            }
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "style": "scss"
        }
      }
    },
    "assets": {
      "projectType": "library",
      "root": "libs/assets",
      "sourceRoot": "libs/assets/src",
      "prefix": "flow-workspace",
      "architect": {},
      "schematics": {}
    },
    "utilities": {
      "projectType": "library",
      "root": "libs/utilities",
      "sourceRoot": "libs/utilities/src",
      "prefix": "flow-workspace",
      "architect": {
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/utilities/tsconfig.lib.json", "libs/utilities/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/utilities/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/utilities/jest.config.js",
            "tsConfig": "libs/utilities/tsconfig.spec.json",
            "passWithNoTests": true,
            "setupFile": "libs/utilities/src/test-setup.ts"
          }
        }
      },
      "schematics": {}
    },
    "flow": {
      "projectType": "library",
      "root": "libs/flow",
      "sourceRoot": "libs/flow/src",
      "prefix": "flow-lib",
      "architect": {
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/flow/tsconfig.lib.json", "libs/flow/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/flow/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/flow/jest.config.js",
            "tsConfig": "libs/flow/tsconfig.spec.json",
            "passWithNoTests": true,
            "setupFile": "libs/flow/src/test-setup.ts"
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "style": "scss"
        }
      }
    }
  },
  "cli": {
    "defaultCollection": "@nrwl/angular"
  },
  "schematics": {
    "@nrwl/angular:application": {
      "unitTestRunner": "jest",
      "e2eTestRunner": "cypress"
    },
    "@nrwl/angular:library": {
      "unitTestRunner": "jest"
    }
  },
  "defaultProject": "flow-web-app"
}
