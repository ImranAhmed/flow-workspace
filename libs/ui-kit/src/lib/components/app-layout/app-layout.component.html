<clr-main-container class="main-container">
  <clr-header class="header-6">
    <div class="branding"></div>
    <div class="header-actions"></div>
  </clr-header>

  <div class="content-container" *ngIf="activeApp">
    <clr-vertical-nav [clrVerticalNavCollapsible]="true" class="nav-trigger--bottom" [clr-nav-level]="1">
      <clr-vertical-nav-group [attr.title]="nav.name" [class.active]="nav.id === activeModuleId" *ngFor="let nav of activeApp.navigation">
        <clr-icon [attr.shape]="nav.icon" clrVerticalNavIcon></clr-icon>
        {{ nav.name }}
        <clr-vertical-nav-group-children>
          <a [routerLink]="c.href" class="nav-link" [class.active]="c.href === activeUrl" *ngFor="let c of nav.children" clrVerticalNavLink>
            {{ c.name }}
          </a>
        </clr-vertical-nav-group-children>
      </clr-vertical-nav-group>
    </clr-vertical-nav>

    <main class="content-area"></main>
  </div>
</clr-main-container>
