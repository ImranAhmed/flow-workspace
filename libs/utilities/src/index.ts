export * from './lib//models/navigation';
export * from './lib//models/user';
export * from './lib/utilities.module';
