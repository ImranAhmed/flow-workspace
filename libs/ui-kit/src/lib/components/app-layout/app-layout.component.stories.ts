import { APP_BASE_HREF } from '@angular/common';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterModule } from '@angular/router';
import { ClarityModule } from '@clr/angular';
import { array } from '@storybook/addon-knobs';
import { AppLayoutComponent } from './app-layout.component';

const getNavigationMenu = (): any[] => {
  return [
    {
      appName: 'App A',
      id: 'app-a',
      icon: 'objects',
      href: '',
      navigation: [
        {
          name: 'Section A',
          id: 'section-a',
          icon: 'scatter-plot',
          href: '',
          children: [
            {
              name: 'Section A1',
              id: 'section-a1',
              href: 'section-a1',
            },
            {
              name: 'Section A2',
              id: 'section-a2',
              href: '',
            },
          ],
        },
        {
          name: 'Section B',
          id: 'section-b',
          icon: 'box-plot',
          href: '',
          children: [
            {
              name: 'Section B1',
              id: 'section-b1',
              href: '',
            },
            {
              name: 'Section B2',
              id: 'section-b2',
              href: '',
            },
          ],
        },
      ],
    },
  ];
};

const getModuleMetadata = () => {
  return {
    imports: [
      BrowserModule,
      BrowserAnimationsModule,
      ClarityModule,
      RouterModule.forRoot([], { useHash: true }),
    ],
    providers: [{ provide: APP_BASE_HREF, useValue: '/' }],
  };
};
export default {
  title: 'AppLayoutComponent',
};

export const withoutNavigation = () => ({
  moduleMetadata: getModuleMetadata(),
  component: AppLayoutComponent,
  props: {
    navigationMenu: array('navigationMenu', null),
  },
});
export const withNavigation = () => ({
  moduleMetadata: getModuleMetadata(),
  component: AppLayoutComponent,
  props: {
    navigationMenu: array('navigationMenu', getNavigationMenu()),
  },
});
