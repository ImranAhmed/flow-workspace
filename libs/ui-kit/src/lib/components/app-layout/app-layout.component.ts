import { Component, Input, OnInit } from '@angular/core';
import { AppNavigation } from '@flow-workspace/utilities';

@Component({
  selector: 'ui-kit-app-layout',
  templateUrl: './app-layout.component.html',
  styleUrls: ['./app-layout.component.scss'],
})
export class AppLayoutComponent implements OnInit {
  @Input() navigationMenu: AppNavigation[];

  activeApp: AppNavigation;
  activeAppId: string;
  activeUrl: string;

  ngOnInit(): void {
    if (this.navigationMenu) {
      this.activeApp = this.navigationMenu[0];
      this.activeAppId = this.activeApp.id;
    }
  }
}
