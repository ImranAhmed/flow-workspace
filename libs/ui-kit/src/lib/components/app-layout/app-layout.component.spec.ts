import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { ClarityModule, ClrVerticalNav } from '@clr/angular';
import { AppNavigation } from '@flow-workspace/utilities';
import { AppLayoutComponent } from './app-layout.component';

const getNavigationMenu = (): AppNavigation[] => {
  return [
    {
      appName: 'App A',
      id: 'app-a',
      icon: 'objects',
      href: '',
      navigation: [
        {
          name: 'Section A',
          id: 'section-a',
          icon: 'scatter-plot',
          href: '',
          children: [
            {
              name: 'Section A1',
              id: 'section-a1',
              href: 'section-a1',
            },
            {
              name: 'Section A2',
              id: 'section-a2',
              href: '',
            },
          ],
        },
      ],
    },
  ];
};

describe('AppLayoutComponent', () => {
  let component: AppLayoutComponent;
  let fixture: ComponentFixture<AppLayoutComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [AppLayoutComponent],
      imports: [BrowserAnimationsModule, ClarityModule, RouterTestingModule],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AppLayoutComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should contain main container', () => {
    const elements = fixture.nativeElement.querySelectorAll('clr-main-container.main-container');
    expect(elements.length).toBe(1);
  });

  describe('header', () => {
    it('should contain header', () => {
      const elements = fixture.nativeElement.querySelectorAll('clr-header.header-6');
      expect(elements.length).toBe(1);
    });

    it('should contain branding', () => {
      const elements = fixture.nativeElement.querySelectorAll('div.branding');
      expect(elements.length).toBe(1);
    });

    it('should contain header-actions', () => {
      const elements = fixture.nativeElement.querySelectorAll('div.header-actions');
      expect(elements.length).toBe(1);
    });
  });

  describe('without navigation', () => {
    beforeEach(() => {
      fixture = TestBed.createComponent(AppLayoutComponent);
      component = fixture.componentInstance;
      component.navigationMenu = null;
      fixture.detectChanges();
    });

    it('should match snapshot', () => {
      expect(fixture).toMatchSnapshot();
    });

    it('should not contain content container', () => {
      const elements = fixture.nativeElement.querySelectorAll('div.content-container');
      expect(elements.length).toBe(0);
    });
  });

  describe('with navigation', () => {
    beforeEach(() => {
      fixture = TestBed.createComponent(AppLayoutComponent);
      component = fixture.componentInstance;
      component.navigationMenu = getNavigationMenu();
      fixture.detectChanges();
    });

    it('should match snapshot', () => {
      expect(fixture).toMatchSnapshot();
    });

    it('should contain container', () => {
      const elements = fixture.nativeElement.querySelectorAll('div.content-container');
      expect(elements.length).toBe(1);
    });

    it('should contain vertival nav', () => {
      const dir = fixture.debugElement.query(By.directive(ClrVerticalNav));

      const nav = dir.injector.get(ClrVerticalNav);

      expect(nav).toBeDefined();
      expect(nav.collapsible).toBeTruthy();
      expect(nav.hasNavGroups).toBeTruthy();
      expect(fixture.nativeElement.querySelectorAll('clr-vertical-nav.nav-trigger--bottom').length).toBe(1);
    });
  });
});
