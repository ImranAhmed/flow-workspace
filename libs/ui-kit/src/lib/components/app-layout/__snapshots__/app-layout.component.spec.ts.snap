// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayoutComponent with navigation should match snapshot 1`] = `
<ui-kit-app-layout
  activeApp={[Function Object]}
  activeAppId={[Function String]}
  navigationMenu={[Function Array]}
>
  <clr-main-container
    class="main-container"
  >
    <clr-header
      class="header-6 header"
    >
      
      <button
        aria-label="Open"
        class="header-hamburger-trigger ng-star-inserted"
        type="button"
      >
        <span />
      </button>
      <div
        class="branding"
      />
      <div
        class="header-actions"
      />
      
      <div
        class="header-backdrop"
      />
    </clr-header>
    
    <div
      class="content-container ng-star-inserted"
    >
      <clr-vertical-nav
        class="nav-trigger--bottom clr-vertical-nav clr-nav-level-1 has-nav-groups has-icons"
      >
        
        <button
          aria-expanded="true"
          aria-label="Toggle vertical navigation"
          class="nav-trigger ng-star-inserted"
          type="button"
        >
          <clr-icon
            class="nav-trigger-icon"
            dir="left"
            shape="angle-double"
            title="Collapse"
          />
        </button>
        <div
          class="nav-content"
        >
          
          <clr-vertical-nav-group
            class="nav-group ng-tns-c7-0 ng-star-inserted"
            title="Section A"
          >
            <div
              class="nav-group-content"
            >
              <button
                aria-expanded="false"
                aria-label="Toggle vertical navigation group"
                class="nav-group-trigger"
                type="button"
              >
                <clr-icon
                  class="nav-icon"
                  clrverticalnavicon=""
                  shape="scatter-plot"
                />
                <div
                  class="nav-group-text"
                >
                   Section A 
                </div>
                <clr-icon
                  class="nav-group-trigger-icon"
                  dir="right"
                  shape="caret"
                  title="Expand"
                />
              </button>
            </div>
            <div
              class="nav-group-children ng-trigger ng-trigger-clrExpand"
              style="height:0px;overflow-y:hidden;visibility:hidden;0:height;1:overflow-y;2:visibility;"
            >
              <clr-vertical-nav-group-children>
                
                <a
                  class="nav-link ng-star-inserted"
                  clrverticalnavlink=""
                  href="/section-a1"
                  style=""
                >
                  <span
                    class="nav-text"
                  >
                     Section A1 
                  </span>
                </a>
                <a
                  class="nav-link ng-star-inserted"
                  clrverticalnavlink=""
                  href="/"
                  style=""
                >
                  <span
                    class="nav-text"
                  >
                     Section A2 
                  </span>
                </a>
              </clr-vertical-nav-group-children>
            </div>
          </clr-vertical-nav-group>
          
        </div>
      </clr-vertical-nav>
      <main
        class="content-area"
      />
    </div>
  </clr-main-container>
</ui-kit-app-layout>
`;

exports[`AppLayoutComponent without navigation should match snapshot 1`] = `
<ui-kit-app-layout
  navigationMenu="null"
>
  <clr-main-container
    class="main-container"
  >
    <clr-header
      class="header-6 header"
    >
      
      <div
        class="branding"
      />
      <div
        class="header-actions"
      />
      
      <div
        class="header-backdrop"
      />
    </clr-header>
    
  </clr-main-container>
</ui-kit-app-layout>
`;
